### PR TITLE
feat: paypal checkout

### PR DESCRIPTION
**Pull request by FabianGerke**
## Why:

Closes: #ID

## Checklist before requesting a review
- [ ] My code follows the code style of this project.
- [ ] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

File: apps/web/components/PayPal/PayPalExpressButton.vue [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+  });
+
+  if (order?.order?.id) {
+    router.push('/order/success');
+  } else {
+    router.push('/order/failed');
+  }
+};
+
+// eslint-disable-next-line sonarjs/cognitive-complexity
```
Comment (ksted) [inline apps/web/components/PayPal/PayPalExpressButton.vue:54]: **suggestion**: Reduce the complexity by extracting the `try` functionality.

File: apps/web/components/PayPal/PayPalExpressButton.vue [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+            },
+          });
+
+          if (button.isEligible() && paypalButton.value) {
+            button.render('#' + paypalButton.value?.id);
+          }
+        }
+      });
+    } catch {
+      // console.error('failed to render the PayPal Buttons', error);
```
Comment (ksted) [inline apps/web/components/PayPal/PayPalExpressButton.vue:101]: **question**: Can we replace the commented out console error with a TODO of how we ultimately intend to handle the error?

File: apps/web/composables/usePayPal/types.ts [hunk 3]
```diff
@@ -0,0 +1,26 @@
+import { PayPalNamespace } from '@paypal/paypal-js';
+import {
+  PayPalCreateOrder,
+  PayPalApproveOrder,
+  PayPalExecutePayment,
+  PayPalExecuteParams,
+} from '@plentymarkets/plentymarkets-sdk/packages/api-client/server';
+
+export interface UsePayPalState {
+  PayPalScript: PayPalNamespace | null;
```
Comment (ksted) [inline apps/web/composables/usePayPal/types.ts:10]:
```suggestion
  paypalScript: PayPalNamespace | null;
```
Is there a reason not to camel case here?

File: apps/web/composables/usePayPal/usePayPal.ts [hunk 4]
```diff
@@ -0,0 +1,103 @@
...
+export const usePayPal: UsePayPalMethodsReturn = () => {
+  const state = useState<UsePayPalState>('usePayPal', () => ({
+    PayPalScript: null,
+    order: null,
+  }));
+
+  /**
+   * @description Function for fetching the category tree.
+   * @example
+   * loadScript(currency: string);
```
Comment (ksted) [inline apps/web/composables/usePayPal/usePayPal.ts:29]: **todo**: Update annotations.

File: apps/web/composables/usePayPal/usePayPal.ts [hunk 5]
```diff
@@ -0,0 +1,103 @@
...
+    if (paypalGetters.getClientId()) {
+      try {
+        state.value.PayPalScript = await loadPayPalScript({
+          clientId: paypalGetters.getClientId() ?? '',
+          currency: currency,
+          dataPartnerAttributionId: 'plentysystemsAG_Cart_PPCP',
+        });
+        return state.value.PayPalScript;
+      } catch {
+        // console.error('failed to load the PayPal JS SDK script', error);
```
Comment (ksted) [inline apps/web/composables/usePayPal/usePayPal.ts:41]: **question**: Can we replace the commented out console error with a TODO of how we ultimately intend to handle the error?

File: apps/web/composables/usePayPal/usePayPal.ts [hunk 6]
```diff
@@ -0,0 +1,103 @@
...
+      }
+    }
+
+    return null;
+  };
+
+  /**
+   * @description Function for creating a PayPal transaction.
+   * @example
+   * createOrder(fundingSource: string);
```
Comment (ksted) [inline apps/web/composables/usePayPal/usePayPal.ts:51]: **thought**: The `fundingSource` being a string reads more like an `@param` to me. My understanding is that `@example` should show a concrete use case or describe how to use the function without a code sample.

File: apps/web/composables/usePaymentMethods/usePaymentMethods.ts [hunk 7]
```diff
@@ -7,6 +7,7 @@ import { UsePaymentMethodsReturn, UsePaymentMethodsState, FetchPaymentMethods, S
  * @description Composable for getting payment methods.
  * @example
  * const { data, loading, fetchPaymentMethods, savePaymentMethod } = usePaymentMethods();
+ * TODO: Remove .selected
```
Comment (ksted) [inline apps/web/composables/usePaymentMethods/usePaymentMethods.ts:10]: **question**: What is this about?

File: apps/web/integration.config.ts [hunk 8]
```diff
@@ -0,0 +1,19 @@
...
+    };
+  };
+}
+
+export default {
+  payment: {
+    paypal: {
+      paymentId: 6001,
+      clientId: 'Ab_wQoMAfzuqCrl4gVfYvkNHmBS_s_rQKMafFJrArKJ4GZU8nbSIn53v4Q8ZZfoHR01kxnjkDF4yVLAv',
+      merchantId: '999JUBU6WMRWU',
```
Comment (ksted) [inline apps/web/integration.config.ts:16]: **question**: Where can we add the partner attribution ID?

**Review by ksted — CHANGES_REQUESTED** [review]

File: apps/web/integration.config.ts (hunk 8 above)
Comment (FabianGerke) [reply apps/web/integration.config.ts:16]: usePayPal.ts -> loadScript

**Review by ksted — APPROVED** [review]

**Pull request merged**